### PR TITLE
Feat/deposit and distribute

### PR DESCRIPTION
**Pull request by JaredBorders**
Add way for users to batch deposit margin and open/modify/close positions

## Description
* Create `depositAndDistribute()`
* Test (unit and integration) `depositAndDistribute()`

## Related issue(s)
Closes https://github.com/Kwenta/margin-manager/issues/12

## Motivation and Context
Enhancement Feature for Margin Accounts

### CODE REVIEW

File: contracts/MarginBase.sol [hunk 1]
```diff
@@ -312,7 +312,17 @@ contract MarginBase is MinimalProxyable, OpsReady, IMarginBaseTypes {
...
-    // @TODO https://github.com/Kwenta/margin-manager/issues/12
+    /// @notice accept a deposit amount and open new
+    /// futures market position(s) all in a single tx
+    /// @param _amount: amount of marginAsset to deposit into marginBase account
+    /// @param _newPositions: an array of UpdateMarketPositionSpec's used to modify active market positions
+    function depositAndDistribute(
+        uint256 _amount,
+        UpdateMarketPositionSpec[] memory _newPositions
+    ) external onlyOwner {
+        deposit(_amount);
```
Comment (JChiaramonte7) [inline contracts/MarginBase.sol:323]: can you double check this? deposit has an onlyOwner modifier, in deposit() ,is `msg.sender` == account owner or does it equal this contract

Comment (JaredBorders) [reply contracts/MarginBase.sol:323]: msg.sender is still EOA. I will add another test exhibiting that behavior though just to be clear

Comment (JaredBorders) [reply contracts/MarginBase.sol:323]: Added test!

File: test/integration/margin.behavior.ts [hunk 2]
```diff
@@ -807,4 +808,24 @@ describe("Integration: Test Cross Margin", () => {
...
+            },
+        ];
+
+        // execute trade
+        const tx = marginAccount
+            .connect(account1)
+            .depositAndDistribute(TEST_VALUE, newPosition);
+
+        await expect(tx).to.be.revertedWith("Ownable: caller is not the owner")
+    });
```
Comment (JChiaramonte7) [inline test/integration/margin.behavior.ts:830]: Really appreciate the test, but can we move this to the unit test section? Not really an integration test.

Foundry has a `prank` cheat code we can use or we can just make a folder for hardhat unit tests.

**Review by JChiaramonte7 — CHANGES_REQUESTED** [review]

File: test/integration/margin.behavior.ts (hunk 2 above)
Comment (JaredBorders) [reply test/integration/margin.behavior.ts]: Changed!

**Review by JChiaramonte7 — APPROVED** [review]

**Pull request merged**